static/js/message_list_view.js

```diff
@@ -56,6 +56,40 @@ function same_recipient(a, b) {
     return util.same_recipient(a.msg, b.msg);
 }
 
+function message_was_resolved(message) {
+    if (message.topic === undefined) {
+        return false;
+    }
+
+    // if the message topic starts with the resolved topic prefix, the
+    // topic is just resolved, nothing more.
+    if (message.topic.startsWith(message_edit.RESOLVED_TOPIC_PREFIX)) {
+        return true;
+    }
+
+    // To find unresolved situation we will have to dig information
+    // from edit_history. We use prev_topic and consider message resolved/unresolved when:
+    // 1. are either same as current topic
+    // 2. or they start with ✔
+    if (message.edit_history !== undefined) {
+        const last_edit = message.edit_history.at(-1);
+        let prev_topic;
+        if (util.get_edit_event_prev_topic(last_edit) !== undefined) {
+            prev_topic = util.get_edit_event_prev_topic(last_edit);
+        } else {
+            prev_topic = last_edit.prev_topic;
+        }
+        if (
+            prev_topic !== undefined &&
+            (prev_topic === message.topic ||
+                prev_topic.startsWith(message_edit.RESOLVED_TOPIC_PREFIX))
+        ) {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
I think this implementation can't work, since we're just looking at the last edit history entry.  You want to have a loop structure, presumably modifying the existing loop we have to display MOVED.